### ISSUE TITLE
Dose anyone can give an example of FCN using keras?

### ISSUE THREAD

**Issue by meanmee** [fchollet/keras#1986]
I just have some trouble in using ConVNet for semantic Segmentation. Since I am not sure about how to create the data(like feeding the Pascal VOC data to the Segmentation model coded by Keras ) and the loss function.
reference: https://gist.github.com/longjon/16db1e4ad3afc2614067#file-readme-md

**Comment by lukovkin** [comment]
We are working on Undecimated FCN for sequences, inspired by paper (arXiv:1508.00317v1 [
stat.ML]), which is based on works on Semantic Segmentation.
Work is in progress, repo is here - https://github.com/lukovkin/ufcnn-keras.
Our subject is a bit different, so we saw the model you mentioned, but not going to reproduce it exactly.

**Comment by moo169** [comment]
Could you share me a FCN Source code by keras for me. Thank u very much.  My email is "liuy2c@163.com"

**Comment by ahundt** [comment]
@meanmee @moo169 
These items & repositories are also relevant to FCN:
 - official [keras-contrib](https://github.com/farizrahman4u/keras-contrib) repository which is the upstream source for Keras, and has a [DensenetFCN](https://github.com/farizrahman4u/keras-contrib/blob/master/keras_contrib/applications/densenet.py) implementation
 - https://github.com/farizrahman4u/keras-contrib/issues/47 incorporating coco + voc 2012
 - https://github.com/farizrahman4u/keras-contrib/issues/59 dealing with gaps in FCN training performance
 - https://github.com/farizrahman4u/keras-contrib/issues/63 training on other FCN networks
 - [Keras-FCN](https://github.com/aurora95/Keras-FCN), which I was planning to adapt for a merge into keras-contrib also has a SegDataGenerator implementation with some of the features you are looking for in your comments, plus additional models and experimental support for coco.
 - https://github.com/0bserver07/One-Hundred-Layers-Tiramisu
